.github/workflows/main.yml

```diff
@@ -67,11 +67,17 @@ jobs:
     steps:
       - name: Check out source code
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 32  # This is necessary to get the commits
```

Comment:
This is not related to looking for changed files.  Maybe it should just be part of the first commit that adds the linter job?

Reply:
I think `CHANGED_FILES` will be dependent on how many commits have been fetched.
I am experimenting with it by providing the `fetch-depth: 2`. But it seems like it doesn't change anything.

Reply:
It won't change anything if the PR has no more than 2 commits on top of the destination branch.  But most PRs have more than that, and rarely more than 20 so that's why I picked 32 as a "round" number.  If some commits are not checked out, then they won't have annotations.

Reply:
So, as an experiment, I tried `fetch-depth: 2` and `fetch-depth: 3` in https://github.com/kernelci/kernelci-api/pull/217. Instead of checking out commits based on provided value, it gets all the commits in the checkout and shows annotations for all modified files.

Reply:
OK yes so it is _also_ related to `CHANGED_FILES` as if the depth is less than the number of commits in the PR then the diff revision is not valid.

As we found out today, `--depth` is the number of extra commits to fetch in the history.  So with `--depth=1` you get the current revision and one history commit so 2 commits.

Reply:
> As we found out today, `--depth` is the number of extra commits to fetch in the history. So with `--depth=1` you get the current revision and one history commit so 2 commits.

Actually, that is true for `--depth=2`.
Also, I added another linter error to https://github.com/kernelci/kernelci-api/pull/217 to confirm if the annotations are shown based on the provided depth. It is working fine.

Reply:
I'm pretty sure there were 3 commits with `--depth=2` but maybe that's because one was a merge commit.  Either way, I think that's resolved now.